homeassistant/components/shelly/entity.py

```diff
@@ -420,13 +421,16 @@ def __init__(
         block: Block,
         attribute: str,
         description: BlockEntityDescription,
+        domain: str | None = None,
     ) -> None:
         """Initialize sensor."""
         super().__init__(coordinator, block)
         self.attribute = attribute
         self.entity_description = description
 
-        self._attr_unique_id: str = f"{super().unique_id}-{self.attribute}"
+        self._attr_unique_id: str = f"{super().unique_id}"
+        if domain != Platform.SWITCH:
+            self._attr_unique_id += f"-{self.attribute}"
```

Comment:
We shouldn't have platform specific logic in the base class. We should instead override the value in the sub class

Reply:
Done